shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -45,18 +100,185 @@ impl RedisConfig {
+        // Replace cass_request messages with cassandra responses in place.
+        // We reuse the vec like this to save allocations.
+        let mut messages_redis_response_iter = messages_redis_response.into_iter();
+        /* there is a redis response for each statement in a CassandraMessage so we have to map
+        the redis responses back to the cassandra requests
+         */
+
+        for cass_request in cassandra_messages.iter_mut() {
+            // the responses for this request
+            let cassandra_result: Result<CassandraResult, CacheableState> =
+                if let Some(Frame::Cassandra(frame)) = cass_request.frame() {
+                    let queries = frame.operation.queries();
+                    if queries.len() != 1 {
+                        Err(CacheableState::Err {
+                            reason: "Cacheable Cassandra query must be only one statement".into(),
+                        })
+                    } else if let Some(mut redis_response) = messages_redis_response_iter.next() {
+                        match redis_response.frame() {
+                            Some(Frame::Redis(redis_frame)) => {
+                                match redis_frame {
+                                    RedisFrame::SimpleString(_) => Err(CacheableState::Err {
+                                        reason: "Redis returned a simple string".into(),
+                                    }),
+                                    RedisFrame::Error(e) => {
+                                        return Err(CacheableState::Err {
+                                            reason: e.to_string(),
+                                        })
+                                    }
+                                    RedisFrame::Integer(_) => Err(CacheableState::Err {
+                                        reason: "Redis returned an int value".into(),
+                                    }),
+                                    RedisFrame::BulkString(redis_bytes) => {
+                                        // Redis response contains serialized version of result struct from CassandraOperation::Result( result )
+                                        let mut cursor = Cursor::new(redis_bytes.as_ref());
+                                        let answer =
+                                            CassandraResult::from_cursor(&mut cursor, Version::V4);
+                                        if let Ok(result) = answer {
+                                            Ok(result)
+                                        } else {
+                                            Err(CacheableState::Err {
+                                                reason: answer.unwrap_err().to_string(),
+                                            })
+                                        }
+                                    }
+                                    RedisFrame::Array(_) => Err(CacheableState::Err {
```

Comment:
RedisFrame::Array, RedisFrame::SimpleString and RedisFrame::Integer cases could probably just be written as a generic `_ =>` case.
At the very least grouping them next to each other rather than spread out amongst the other cases would make it easier to follow.